src/source/Ice/SocketConnection.c

```diff
@@ -96,54 +82,67 @@ STATUS freeSocketConnection(PSocketConnection* ppSocketConnection)
     return retStatus;
 }
 
-STATUS socketConnectionInitSecureConnection(PSocketConnection pSocketConnection, BOOL isServer)
+static STATUS tlsSessionOutBoundPacket(UINT64 customData, PBYTE pBuffer, UINT32 bufferLen)
 {
-    ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
+    PSocketConnection pSocketConnection = NULL;
+    CHK(customData != 0, STATUS_NULL_ARG);
 
-    CHK(pSocketConnection != NULL, STATUS_NULL_ARG);
-
-    pSocketConnection->pSslCtx = SSL_CTX_new(SSLv23_method());
-
-    CHK(pSocketConnection->pSslCtx != NULL, STATUS_SSL_CTX_CREATION_FAILED);
-
-    SSL_CTX_set_read_ahead(pSocketConnection->pSslCtx, 1);
-    SSL_CTX_set_verify(pSocketConnection->pSslCtx, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, certificateVerifyCallback);
+    pSocketConnection = (PSocketConnection) customData;
+    retStatus = socketSendDataWithRetry(pSocketConnection, pBuffer, bufferLen, NULL, NULL);
```

Comment:
simply do CHK_STATUS(socketSendData());

Next time we add more data below could intro a bug

Reply:
makes sense.